notebooks/03.02. Supervised Learning.Rmd

```diff
@@ -22,118 +27,177 @@ We will use the following classification algorithms:
-onehot_test$damage_grade <- as.factor(onehot_test$damage_grade)
+This method is a combination of tree predictors such that each tree depends on the values of a random vector sampled independently and with the same distribution for all trees in the forest. Ranger is a fast implementation of Random Forest for high-dimensional data.
 
-# Predict the test dataset
-onehot_test$damage_grade_pred <- predict(rf_model, newdata = onehot_test)
-onehot_test$damage_grade_pred <- as.factor(onehot_test$damage_grade_pred)
+Tuning parameters:
+- `mtry` (#Randomly Selected Predictors)
+- `splitrule` (Splitting Rule)
+- `min.node.size` (Minimal Node Size)
 
-# Ensure that both 'damage_grade' and 'damage_grade_pred' have the same levels
-levels(onehot_test$damage_grade_pred) <- levels(onehot_test$damage_grade)
+```{r}
+# Define the tuning grid
+rf_tuneGrid <- expand.grid(.mtry = c(2, 3, 4, 5, 6, 7, 8, 9, 10),
+                        .splitrule = "gini",
+                        .min.node.size = c(1, 5, 10, 15, 20))
 
-# Predict with probabilities
-onehot_test$probabilities <- predict(rf_model, newdata = onehot_test, type = "prob")
+# Define the model and predictions paths
+rf_model_path <- "../models/rf_onehot_model.RData"
+rf_predictions_path <- "../data/predictions/rf_onehot_predictions.csv"
 
-# Save the predictions
-write.csv(onehot_test, "../data/predictions/rf_onehot_predictions.csv", row.names = FALSE)
+# Apply the algorithm
+rf_model <- apply_algorithm("ranger", rf_tuneGrid, onehot_train, onehot_test, rf_model_path, rf_predictions_path)
 ```
 
-#### Confusion Matrix
-Now we can evaluate the model using the confusion matrix. 
+Now we can evaluate the model using the confusion matrix and other metrics.
 
 ```{r}
 # Load the predictions
-rf_onehot_predictions <- read.csv("../data/predictions/rf_onehot_predictions.csv")
-rf_onehot_predictions$damage_grade_pred <- as.factor(rf_onehot_predictions$damage_grade_pred)
-
-# Save the confusion matrix
-confusion_matrix <- confusionMatrix(rf_onehot_predictions$damage_grade_pred, onehot_test$damage_grade, mode = "everything")
-confusion_matrix
-
-confusion_matrix_table <- as.data.frame(confusion_matrix$byClass)
-## Rename row from Class: Grade_1 to Grade_1
-rownames(confusion_matrix_table)[rownames(confusion_matrix_table) == "Class: Grade_1"] <- "Grade_1"
-rownames(confusion_matrix_table)[rownames(confusion_matrix_table) == "Class: Grade_2"] <- "Grade_2"
-rownames(confusion_matrix_table)[rownames(confusion_matrix_table) == "Class: Grade_3"] <- "Grade_3"
-```
+rf_onehot_predictions <- read.csv(rf_predictions_path)
 
-#### ROC and AUC Multiclass Curves
-The ROC curve is a graphical plot that illustrates the diagnostic ability of a binary classifier system as its discrimination threshold is varied. The ROC curve is created by plotting the true positive rate (TPR) against the false positive rate (FPR) at various threshold settings.
+# Get the metrics
+metrics <- get_prediction_metrics(onehot_test, rf_onehot_predictions)
```

Comment:
I recommend that you use different names for each dataset and algorithm, so that we can compare all the results once we are done (e.g. rf_onehot_metrics).

Reply:
Great suggestion!